python/kfserving/kfserving/api/create_credentials.py

```diff
@@ -0,0 +1,226 @@
+# Copyright 2019 The Kubeflow Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import json
+import configparser
+from os.path import expanduser
+
+from kubernetes import client
+from ..constants import constants
+
+logger = logging.getLogger(__name__)
+
+
+def create_gcs_credentials(namespace, credentials_file=None, service_account=None):
+    '''
+    Create GCP Credentails (secret and service account) with GCP credentials file.
+    Args:
+        credentials_file(str): The path for the gcp credentials file (Optional).
+        service_account(str): The name of service account(Optional). If the service_account
+                              is specified, will attach created secret with the service account,
+                              otherwise will create new one and attach with created secret.
+    return:
+        gcs_sa_name(str): The name of created service account.
+    '''
+
+    if credentials_file is None:
+        credentials_file = constants.GCS_DEFAULT_CREDS_FILE
+
+    with open(expanduser(credentials_file)) as f:
+        gcs_creds_content = f.read()
+
+    # Try to get GCP creds file name from configmap, set default value then if cannot.
+    gcs_creds_file_name = get_creds_name_from_config_map(
+        'gcsCredentialFileName')
+    if not gcs_creds_file_name:
+        gcs_creds_file_name = constants.GCS_CREDS_FILE_DEFAULT_NAME
+
+    string_data = {gcs_creds_file_name: gcs_creds_content}
+
+    secret_name = create_secret(
+        namespace=namespace, string_data=string_data)
+
+    if 'service_account' is None:
+        sa_name = create_service_account(
+            secret_name=secret_name,
+            namespace=namespace)
+    else:
+        sa_name = patch_service_account(
+            secret_name=secret_name,
+            namespace=namespace,
+            sa_name=service_account)
+
+    return sa_name
+
+
+def create_s3_credentials(namespace, **kwargs):  # pylint: disable=too-many-locals
```

Comment:
here we can use explicit args instead of variable args ?

Reply:
Done.